framework/src/base/NonlinearSystem.C

```diff
@@ -870,6 +870,7 @@ NonlinearSystem::enforceNodalConstraintsJacobian(SparseMatrix<Number> & jacobian
       }
     }
   }
+  jacobian.close();
```

Comment:
You only need a `close()` when switching betweening adding and setting or modifying (via either adding or setting) and reading.

In this case it is probably a good idea because is routine is "finished" with the vector and it may be read from later.